### PR TITLE
OWLS-101023: Add Cluster conditions to Cluster status

### PR DESCRIPTION
**Pull request by lennyphan**
Added to Cluster Resource status:

- Cluster Conditions (Available and Completed) 
- Cluster events related to Available and Completed conditions

Example Cluster Resource status with conditions and events:
Spec:
...
Spec:
  Cluster Name:  cluster-1
  Domain UID:    sample-domain1
  Replicas:      2
Status:
  Cluster Name:  cluster-1
  Conditions:
    Last Transition Time:  2022-08-10T02:44:13.278435Z
    Status:                True
    Type:                  Available
    Last Transition Time:  2022-08-10T02:44:13.278526Z
    Status:                True
    Type:                  Completed
  Maximum Replicas:        5
  Minimum Replicas:        0
  Ready Replicas:          2
  Replicas:                2
  Replicas Goal:           2
Events:
  Type     Reason              Age                    From               Message
  ----     ------              ----                   ----               -------
  Warning  ClusterIncomplete   8m18s                  weblogic.operator  Cluster cluster-1 is incomplete for one or more of the following reasons: there are failures detected, there are pending server shutdowns, or not all servers expected to be running are ready and at their target image, auxiliary images, restart version, and introspect version.
  Warning  ClusterUnavailable  8m18s                  weblogic.operator  Cluster cluster-1 is unavailable: an insufficient number of its servers that are expected to be running are ready.";
  Normal   ClusterAvailable    6m32s (x3 over 8m14s)  weblogic.operator  Cluster cluster-1 is available: a sufficient number of its servers have reached the ready state.
  Normal   ClusterCompleted    6m32s (x3 over 8m14s)  weblogic.operator  Cluster cluster-1 is complete because all of the following are true: there is no failure detected, there are no pending server shutdowns, and all servers expected to be running are ready and at their target image, auxiliary images, restart version, and introspect version.

---------------------------------------------------------
Example Domain Resource status:
Spec:
...
Status:
  Clusters:
    Cluster Name:  cluster-1
    Conditions:
      Last Transition Time:  2022-08-10T02:50:13.339441Z
      Status:                True
      Type:                  Available
      Last Transition Time:  2022-08-10T02:50:13.339488Z
      Status:                True
      Type:                  Completed
    Maximum Replicas:        5
    Minimum Replicas:        0
    Ready Replicas:          2
    Replicas:                2
    Replicas Goal:           2
  Conditions:
    Last Transition Time:  2022-08-10T02:46:13.267970Z
    Status:                True
    Type:                  Available
    Last Transition Time:  2022-08-10T02:46:13.267741Z
    Status:                True
    Type:                  Completed
  Observed Generation:     1
  Replicas:                2

### CODE REVIEW

File: operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java [hunk 1]
```diff
@@ -675,6 +677,27 @@ public Conditions(DomainStatus status) {
 
         void apply() {
           conditionList.forEach(newCondition -> addCondition(status, newCondition));
+          setClusterStatusConditions(status.getClusters());
+        }
+
+        private void setClusterStatusConditions(List<ClusterStatus> clusterStatuses) {
```
Comment (russgold) [inline operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java:683]: Please reverse this logic. Each ClusterCheck contains the name of its cluster, so you iterate over the checks and then ask for the corresponding status. This would let you move the `addClusterConditions` code into ClusterCheck and not need to test if it's null.

**Review by russgold — COMMENTED** [review]
I don't understand why we have a duplicate set of condition types for the clusters. Aren't they necessarily a subset of the condition types for the domain? Are there any exceptions to that?

**Review by rjeberhard — APPROVED** [review]

**Review by doxiao — APPROVED** [review]
LGTM.
